src/spetlr/etl/loaders/stream_loader.py

```diff
@@ -0,0 +1,117 @@
+import uuid as _uuid
+
+from pyspark.sql import DataFrame
+from pyspark.sql.streaming import DataStreamWriter
+
+from spetlr.etl import Loader
+from spetlr.exceptions import (
+    NeedTriggerTimeWhenProcessingType,
+    NotAValidStreamTriggerType,
+    SparkVersionNotSupportedForSpetlrStreaming,
+    UnknownStreamOutputMode,
+)
+from spetlr.spark import Spark
+
+
+class StreamLoader(Loader):
+    def __init__(
+        self,
+        *,
+        loader: Loader,
+        checkpoint_path: str = None,
+        options_dict: dict = None,
+        trigger_type: str = "availablenow",
+        trigger_time_seconds: int = None,
+        outputmode: str = "update",
+        query_name: str = None,
+        await_termination: bool = False,
+    ):
+        """
+        checkpoint_path: The location of the checkpoints, <table_name>/_checkpoints
+            The Delta Lake VACUUM function removes all files not managed by Delta Lake
+            but skips any directories that begin with _. You can safely store
+            checkpoints alongside other data and metadata for a Delta table
+            using a directory structure such as <table_name>/_checkpoints
+            See: https://docs.databricks.com/structured-streaming/delta-lake.html
+        location: the location of the delta table (Optional)
+        trigger_type: the trigger type of the stream.
+            See: https://docs.databricks.com/structured-streaming/triggers.html
+        trigger_time: if the trigger has is "processingtime",
+            it should have a trigger time associated
+        """
+
+        super().__init__()
+        self._loader = loader
+        self._options_dict = options_dict
+        self._outputmode = outputmode
+        self._trigger_type = trigger_type
+        self._trigger_time_seconds = trigger_time_seconds
+        self._query_name = query_name or str(
+            _uuid.uuid4().hex
+        )  # Consider if this is smart?
+        self._checkpoint_path = checkpoint_path
+        self._await_termination = await_termination
+        self._validate_checkpoint()
+
+        if Spark.version() < Spark.DATABRICKS_RUNTIME_10_4:
```

Comment:
When having a spark version check here, is it necessary on the readStream() handlers?
I would argue it is needed here because we use the availableNow, but someone could be using the handler.readStream() with there own loader?

Reply:
Good point. I think i'll remove it from the handler